src/bundle/mutable.js

```diff
@@ -0,0 +1,144 @@
+   * Because of lack of sufficient foresight at the time, ES5 unfortunately
+   * specified that a simple assignment to a non-existent property must fail if
+   * it would override a non-writable data property of the same name. (In
+   * retrospect, this was a mistake, but it is now too late and we must live
+   * with the consequences.) As a result, simply freezing an object to make it
+   * tamper proof has the unfortunate side effect of breaking previously correct
+   * code that is considered to have followed JS best practices, if this
+   * previous code used assignment to override.
+   *
+   * To work around this mistake, deepFreeze(), prior to freezing, replaces
+   * selected configurable own data properties with accessor properties which
+   * simulate what we should have specified -- that assignments to derived
+   * objects succeed if otherwise possible.
+   */
+  function beMutable(obj, prop, desc) {
+    if ('value' in desc && desc.configurable) {
+      const { value } = desc;
+
+      // eslint-disable-next-line no-inner-declarations
+      function getter() {
+        return value;
+      }
+
+      // Re-attach the data property on the object so
+      // it can be found by the deep-freeze traversal process.
+      getter.value = value;
+
+      // eslint-disable-next-line no-inner-declarations
+      function setter(newValue) {
+        if (obj === this) {
+          throw new TypeError(
+            `Cannot assign to read only property '${prop}' of object '${obj}'`,
+          );
+        }
+        if (hasOwnProperty.call(this, prop)) {
+          this[prop] = newValue;
+        } else {
+          defineProperty(this, prop, {
+            value: newValue,
+            writable: true,
+            enumerable: desc.enumerable,
+            configurable: desc.configurable,
+          });
+        }
+      }
+
+      defineProperty(obj, prop, {
+        get: getter,
+        set: setter,
+        enumerable: desc.enumerable,
+        configurable: desc.configurable,
+      });
+    }
+  }
+
+  function beMutableProperties(obj) {
+    if (!obj) {
+      return;
+    }
+    const descs = getOwnPropertyDescriptors(obj);
+    if (!descs) {
+      return;
+    }
+    ownKeys(obj).forEach(prop => beMutable(obj, prop, descs[prop]));
+  }
+
+  /**
+   * These properties are subject to the override mistake
+   * and must be converted before freezing.
+   */
+  function repairDataProperties(intrinsics) {
+    const { global: g, anonIntrinsics: a } = intrinsics;
+
+    const toBeRepaired = [
+      g.Object.prototype,
+      g.Array.prototype,
+      g.Boolean.prototype,
+      g.Date.prototype,
+      g.Number.prototype,
+      g.String.prototype,
+
+      g.Function.prototype,
+      a.GeneratorFunction.prototype,
+      a.AsyncFunction.prototype,
+      a.AsyncGeneratorFunction.prototype,
+
+      a.IteratorPrototype,
+      a.ArrayIteratorPrototype,
+
+      g.DataView.prototype,
+
+      a.TypedArray,
+      a.TypedArray.prototype,
```

Comment:
`TypedArray.prototype` not present in `forcedotcom/aura` or `tc39/ses-proposal`, but needed for [user-space `Buffer` package](https://github.com/feross/buffer)

Reply:
If you don't know of an immediate need to include non-prototypes, I would drop all non-prototypes until we come across an actual problem.

Adding `TypedArray.prototype` to the list should be fine.

We should consider a finer grain whitelist, of individual properties rather than just wholesale objects. I doubt we will often run into problems with any primordial properties other than `toString` methods.

Reply:
@erights I believe Buffer.from conflicts with TypedArray.from if not repaired

Reply:
update on `buffer` npm package:
with repairing `TypedArray.prototype` but not `TypedArray`

this code would fail:
```js
r = require('./dist/ses.umd.js').makeSESRootRealm()

code = `
function Buffer () {}

Object.setPrototypeOf(Buffer.prototype, Uint8Array.prototype)
Object.setPrototypeOf(Buffer, Uint8Array)

Buffer.from = () => {}
`
r.evaluate(code)
```

however the `buffer` package actually defines `from` before setting the prototype, in response to [an error reported](https://github.com/feross/buffer/pull/148#issuecomment-274408688) with `chrome 50.0.2661.102` where `from` was not writable, before being made writable again in a later version.

So this doesn't actually affect the `buffer` package, but could affect others using this pattern

Reply:
I included `TypedArray.prototype` for now and removed `TypedArray` . Will see how it fares and report any issues.